### PR TITLE
Updating to include aarch64 and ppc64le

### PR DESCRIPTION
**Pull request by step21**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by step21** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
i didn't know matplotlib was built yet. was it?

**Comment by jakirkham** [comment]
Appears to be.

**Comment by hmaarrfk** [comment]
its because the qt dependency was skipped.

seems like a bad idea ;)

**Comment by hmaarrfk** [comment]
@step21 if you could follow the style of
https://github.com/conda-forge/numpy-feedstock/blob/master/recipe/meta.yaml#L56

and skip the tests.

When you skip them, could you please create issues upstream so that we (Scikit-image) can fix them.

**Comment by step21** [comment]
Ok will do. I guess for now skipping qt is reasonable, as it is only used for gui stuff in matplotlib...? so for example useless on a headless jupyte notebook install.

**Comment by jakirkham** [comment]
> its because the qt dependency was skipped.

These are the ones that use `pyqt`, right?

https://anaconda.org/conda-forge/matplotlib/files

**Comment by step21** [comment]
@jakirkham yes. they are so small because the only add pyqt. https://github.com/conda-forge/matplotlib-feedstock/blob/master/recipe/meta.yaml

**Comment by step21** [comment]
@conda-forge-admin, please rerender

**Comment by step21** [comment]
One question, in the checks I couldn't find the logs from aarch64, only ppc. Are they part of the normal linux logs?

**Comment by hmaarrfk** [comment]
Aarch is built on Drone. Is that what you are looking for?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -60,6 +60,12 @@ requirements:
   run_constrained:
     - pooch >=0.5.2
 
+{% set tests_to_skip = "test_end_points" %}  # [ppc64le or aarch64]
+{% set tests_to_skip = tests_to_skip + " or test_RGB" %}  # [ppc64le or aarch64]
+{% set tests_to_skip = tests_to_skip + " or test_periodic_reference" %}  # [ppc64le or aarch64]
+{% set tests_to_skip = tests_to_skip + " or test_hdx_rgb_roundtrip" %}  # [ppc64le or aarch64]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:66]:
```suggestion
{% set tests_to_skip = tests_to_skip + " or test_hdx_rgb_roundtrip" %}  # [ppc64le or aarch64 or osx]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -60,6 +60,12 @@ requirements:
...
+{% set tests_to_skip = tests_to_skip + " or test_hed_rgb_roundtrip" %}  # [ppc64le or aarch64]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:67]:
```suggestion
{% set tests_to_skip = tests_to_skip + " or test_hed_rgb_roundtrip" %}  # [ppc64le or aarch64 or osx]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -60,6 +60,12 @@ requirements:
   run_constrained:
     - pooch >=0.5.2
 
+{% set tests_to_skip = "test_end_points" %}  # [ppc64le or aarch64]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:63]:
```suggestion
# Issues reported upstream. Follow
# https://github.com/scikit-image/scikit-image/issues/4768
{% set tests_to_skip = "test_end_points" %}  # [ppc64le or aarch64]
```

**Comment by step21** [comment]
> Aarch is built on Drone. Is that what you are looking for?

yes. But it didn't have errors, so potentially could also be removed from the test exclusion.

**Comment by step21** [comment]
@conda-forge-admin, please rerender

**Comment by step21** [comment]
os x was failing because tests_to_skip was only set if not osx and then couldn't be used on osx. Re-ordered to fix.

**Comment by step21** [comment]
After disabling some tests, as advised, on re-running some other tests failed on ppc64le and 1 on windows that also didn’t fail before - so should I disable those too?

**Comment by hmaarrfk** [comment]
yeah, and please open seperate issues for each, copying the error message, and linking to this specifi PR.

**Comment by hmaarrfk** [comment]
one issue per test you skip;.

**Comment by step21** [comment]
Small question, should I also adjust the selectors, so that aarch64 is only excluded if it fails somewhere, not just because ppc64le fails?

**Comment by step21** [comment]
@conda-forge-admin, please rerender

**Comment by step21** [comment]
It seems some of the tests were not actually skipped, despite being marked to be skipped. Can you see whether I referenced them wrong, or is this just a general problem of ppc64le?

**Comment by step21** [comment]
@conda-forge-admin, please rerender

**Comment by step21** [comment]
@conda-forge-admin, please restart ci

**Comment by step21** [comment]
@conda-forge-admin, please rerender and please restart ci

**Comment by step21** [comment]
@conda-forge-admin, please restart ci

**Comment by step21** [comment]
@conda-forge-admin, please restart ci

**Comment by step21** [comment]
(also sorry for the mess of commits -  I hope they can be. squashed)

**Comment by hmaarrfk** [comment]
i know it sounds like busy work, but I would really like to see direct cross references to each issue on scikit-image.

As a developer of the project, these really help us improve.

I presume you are a user of scikit-image, and would like to see that these functions get fixed in the next release.

In terms of distribution through conda-forge, we need to be able to track progress upstream to know when to stop skipping tests.

Please copy, or screenshot the outputs from travis, done, and azure in each issue creation relating to a specific failure.

**Comment by step21** [comment]
You mean cross-reference here on scikit-image? The biggest problem with that is, that sometimes logs do not seem to be available anymore so if I didn't make a screenshot or copy, they are gone. Plus sometimes errors do not happen every run.  Especially on ppc64le it was kinda annoying. Currently my main use is to re-create jupyterlab docker stacks on aarch64, and making conda packages kinda is a nice formulaic thing to balance out others. But yeah ofc I want things to get fixed. If there is anything to fix. I will try to add the rest of the output tomorrow.

**Comment by hmaarrfk** [comment]
Even linking to this Pull Request on Scikit-image is a great help.

Then on every test you skip, referencing the issue on scikit-image would be greatly helpful. That way, a conda-forge maintainer can do a quick heck on the next update to see if the issue has been solved..

I restarted the builds, so hopefully you should have access at least to some broken logs. Lets see how many we can see.

**Comment by hmaarrfk** [comment]
plot twist, tests don't fail....

I know some of these tests fail intermittently, so i'm going to try again.

**Comment by step21** [comment]
Thanks for the update. Today I should really focus on other stuff, so I probably will look into it again tomorrow. / update the issues etc.

**Comment by hmaarrfk** [comment]
Perfectly understood.

I kinda showed one example of the level of detail I would expect in the upstream bug report.

Again, this is really because we know that users are interested in installing on these platforms, and we wish to create a good experience for them.

Thanks for helping push this through.

**Comment by step21** [comment]
Sure. I hope the upstream reports are fine now like this.

**Comment by hmaarrfk** [comment]
I think so. maybe just icnlude them in the comments in `meta.yaml` and this is ready to go once it is green.

**Comment by step21** [comment]
Ok I will put them into meta.yaml. Does 'green' mean 'once the tests are fixed and they are green consistently'? Because it was green in between, but like you said yourself some tests fail intermittently. For working on actual scikit-image code I do not have resources right now.

**Comment by hmaarrfk** [comment]
now that you have documented things, I think we can just keep skipping tests until things pass.

I'm ok with the concept that these are still experimental for these architectures.

**Comment by step21** [comment]
@hmaarrfk OSX build on python 3.6 fails. but the log looks like it was just off by the ci or something. And builds that succeeded on travis, they took really really long. (4 hours) https://travis-ci.com/github/conda-forge/scikit-image-feedstock/jobs/346848532

**Pull request merged**